compressor/base.py

```diff
@@ -117,18 +117,21 @@ def get_filename(self, basename):
         get_filename('css/one.css') -> '/full/path/to/static/css/one.css'
         """
         filename = None
-        # first try finding the file in the root
-        try:
-            # call path first so remote storages don't make it to exists,
-            # which would cause network I/O
-            filename = self.storage.path(basename)
-            if not self.storage.exists(basename):
-                filename = None
-        except NotImplementedError:
-            # remote storages don't implement path, access the file locally
-            if compressor_file_storage.exists(basename):
-                filename = compressor_file_storage.path(basename)
-        # secondly try to find it with staticfiles (in debug mode)
+        # first try finding the file in STATIC_ROOT.
+        # skip this in DEBUG mode as files might be outdated in STATIC_ROOT,
+        # and do the compression based solely on the app's static dirs instead.
```

Comment:
Replace the 3 lines comment with :
 `First try finding the file using the storage class. This is skipped in DEBUG mode as files might be outdated in compressor's final destination (COMPRESS_ROOT) during development`